File: Clase6/SoftProgSolution/SoftProg/src/main/java/pe/edu/pucp/softprog/main/Principal.java
package pe.edu.pucp.softprog.main;
import pe.edu.pucp.softprog.almacen.bo.ProductoBOImpl;
import pe.edu.pucp.softprog.almacen.boi.IProductoBO;
import pe.edu.pucp.softprog.almacen.dao.ProductoDAO;
import pe.edu.pucp.softprog.almacen.impl.ProductoImpl;
import pe.edu.pucp.softprog.almacen.model.Producto;
import pe.edu.pucp.softprog.gestclientes.bo.ClienteBOImpl;
import pe.edu.pucp.softprog.gestclientes.boi.IClienteBO;
import pe.edu.pucp.softprog.gestclientes.dao.ClienteDAO;
import pe.edu.pucp.softprog.gestclientes.impl.ClienteImpl;
import pe.edu.pucp.softprog.gestclientes.model.Categoria;
import pe.edu.pucp.softprog.gestclientes.model.Cliente;
import pe.edu.pucp.softprog.rrhh.bo.AreaBOImpl;
import pe.edu.pucp.softprog.rrhh.bo.EmpleadoBOImpl;
import pe.edu.pucp.softprog.rrhh.boi.IAreaBO;
import pe.edu.pucp.softprog.rrhh.boi.IEmpleadoBO;
import pe.edu.pucp.softprog.rrhh.dao.AreaDAO;
import pe.edu.pucp.softprog.rrhh.dao.EmpleadoDAO;
import pe.edu.pucp.softprog.rrhh.impl.AreaImpl;
import pe.edu.pucp.softprog.rrhh.impl.EmpleadoImpl;
import pe.edu.pucp.softprog.rrhh.model.Area;
import pe.edu.pucp.softprog.ventas.bo.OrdenVentaBOImpl;
import pe.edu.pucp.softprog.ventas.boi.IOrdenVentaBO;
import pe.edu.pucp.softprog.ventas.dao.OrdenVentaDAO;
import pe.edu.pucp.softprog.ventas.impl.OrdenVentaImpl;
import pe.edu.pucp.softprog.ventas.model.OrdenVenta;
import pe.edu.pucp.softprog.ventas.model.LineaOrdenVenta;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.ResourceBundle;
import pe.edu.pucp.softprog.config.DBManager;
import pe.edu.pucp.softprog.rrhh.model.Empleado;

public class Principal {
    public static void main(String[] args) throws Exception{
        SimpleDateFormat formato = new SimpleDateFormat("dd-MM-yyyy");

        IAreaBO areaBO = new AreaBOImpl();
        Area area = areaBO.buscarPorId(1);
        Empleado empleado = new Empleado("16527933",
                "MANUEL","TUPIA",'M',
                formato.parse("01-03-1977"),area,
                "JEFE DE VENTAS",1500.00);

        IEmpleadoBO empleadoBO = new EmpleadoBOImpl();
        int resultado = empleadoBO.insertar(empleado);
        if(resultado!=0)
            System.out.println("Se ha registrado con exito");
        resultado = empleadoBO.modificar(empleado);
        if(resultado!=0)
            System.out.println("Se ha modificado con exito");
        List<Empleado> empleados = empleadoBO.listarTodos();
        for(Empleado emp : empleados){
            System.out.println(emp.getNombre());
        }
        empleado = empleadoBO.buscarPorId(empleado.getIdPersona());
        resultado = empleadoBO.eliminar(empleado.getIdPersona());

        Cliente cliente = new Cliente("76276541","CAROLINA",
                "CACERES",'F',formato.parse("10-10-1988"),1800.00, Categoria.STANDARD);
        IClienteBO clienteBO = new ClienteBOImpl();
        resultado = clienteBO.insertar(cliente);
        if(resultado!=0)
            System.out.println("Se ha insertado el cliente");
        resultado = clienteBO.modificar(cliente);
        if(resultado!=0)
            System.out.println("Se ha modificado el cliente");
        List<Cliente> clientes = clienteBO.listarTodos();
        for(Cliente c : clientes)
            System.out.println(c.getNombre());

        IProductoBO productoBO = new ProductoBOImpl();
        List<Producto> productos = productoBO.listarTodos();
        for(Producto p : productos){
            System.out.println(p);
        }

        OrdenVenta ov = new OrdenVenta();
        ov.setCliente(cliente);
        ov.setEmpleado(empleado);
        ov.setFechaHora(new Date());

        LineaOrdenVenta lov1 = new LineaOrdenVenta();
        lov1.setProducto(productos.get(0));
        lov1.setCantidadUnidades(2);

        LineaOrdenVenta lov2 = new LineaOrdenVenta();
        lov2.setProducto(productos.get(2));
        lov2.setCantidadUnidades(3);

        ov.agregarLineaOrdenVenta(lov1);
        ov.agregarLineaOrdenVenta(lov2);

        ov.calcularSutotalesYTotal();

        IOrdenVentaBO ordenVentaBO = new OrdenVentaBOImpl();
        resultado = ordenVentaBO.insertar(ov);
        if(resultado!=0)
            System.out.println("La orden de venta se ha registrado");

    }
}

File: Clase12/ProyectosNET/SoftProgSolution/SoftProg/Principal.cs
using Microsoft.Extensions.Configuration;
using MySql.Data.MySqlClient;
using SoftProgDBManager;
using SoftProgModel.Almacen;
using SoftProgModel.GestClientes;
using SoftProgModel.RRHH;
using SoftProgModel.Ventas;
using SoftProgNegocio.Almacen.BO;
using SoftProgNegocio.Almacen.BOI;
using SoftProgNegocio.GestClientes.BO;
using SoftProgNegocio.GestClientes.BOI;
using SoftProgNegocio.RRHH.BO;
using SoftProgNegocio.RRHH.BOI;
using SoftProgNegocio.Ventas.BO;
using SoftProgNegocio.Ventas.BOI;
using SoftProgPersistencia.RRHH.DAO;
using SoftProgPersistencia.RRHH.Impl;

public class Principal
{
    public static void Main(String[] args)
    {
        Area area = new Area("RECURSOS HUMANOS");
        System.Console.WriteLine(area.Nombre);

        IConfiguration configuration = new ConfigurationBuilder()
            .SetBasePath(Directory.GetCurrentDirectory())
            .AddJsonFile("appsettings.json")
            .Build();
        string connectionString = configuration.GetConnectionString("MySqlConnection");

        System.Console.WriteLine(connectionString);

        DBManager.Initialize(connectionString);

        IAreaBO areaBO = new AreaBOImpl();

        areaBO.insertar(area);

        List<Area> areas = areaBO.listarTodos();

        foreach (Area a in areas)
            System.Console.WriteLine(a);

        for (int i = 0; i < areas.Count; i++)
            System.Console.WriteLine(areas[i]);

        Empleado emp1 = new Empleado();
        emp1.Nombre = "HUGO";
        emp1.ApellidoPaterno = "VILLANUEVA";
        emp1.DNI = "98762211";
        emp1.Area = areas[0];
        emp1.Genero = 'M';
        emp1.FechaNacimiento = DateTime.Now;
        emp1.Cargo = "VENDEDOR";
        emp1.Sueldo = 2400.00;

        IEmpleadoBO empleadoBO = new EmpleadoBOImpl();
        empleadoBO.insertar(emp1);

        List<Empleado> empleados = empleadoBO.listarTodos();

        System.Console.WriteLine("Lectura compleada..");

        IProductoBO productoBO = new ProductoBOImpl();
        List<Producto> productos = productoBO.listarTodos();

        IClienteBO clienteBO = new ClienteBOImpl();
        List<Cliente> clientes = clienteBO.listarTodos();

        OrdenVenta ov = new OrdenVenta();

        LineaOrdenVenta lov1 = new LineaOrdenVenta();
        lov1.Producto = productos[0];
        lov1.CantidadUnidades = 2;
        lov1.Subtotal = lov1.Producto.Precio * lov1.CantidadUnidades;

        LineaOrdenVenta lov2 = new LineaOrdenVenta();
        lov2.Producto = productos[1];
        lov2.CantidadUnidades = 3;
        lov2.Subtotal = lov2.Producto.Precio * lov2.CantidadUnidades;

        ov.LineasOrdenVenta.Add(lov1);
        ov.LineasOrdenVenta.Add(lov2);

        ov.Total = lov1.Subtotal + lov2.Subtotal;

        ov.Cliente = clientes[0];
        ov.Empleado = empleados[0];

        IOrdenVentaBO ordenVentaBO = new OrdenVentaBOImpl();
        ordenVentaBO.insertar(ov);

        System.Console.WriteLine("Se ha registrado la ov...");
    }
}